src/qt/transactiondesc.cpp

```diff
@@ -259,22 +259,45 @@ QString TransactionDesc::toHTML(interfaces::Node& node, interfaces::Wallet& wall
         if (r.first == "Message")
             strHTML += "<br><b>" + tr("Message") + ":</b><br>" + GUIUtil::HtmlEscape(r.second, true) + "<br>";
 
-#ifdef ENABLE_BIP70
     //
     // PaymentRequest info:
     //
     for (const std::pair<std::string, std::string>& r : orderForm)
     {
         if (r.first == "PaymentRequest")
         {
+#ifdef ENABLE_BIP70
             PaymentRequestPlus req;
             req.parse(QByteArray::fromRawData(r.second.data(), r.second.size()));
             QString merchant;
             if (req.getMerchant(PaymentServer::getCertStore(), merchant))
                 strHTML += "<b>" + tr("Merchant") + ":</b> " + GUIUtil::HtmlEscape(merchant) + "<br>";
+#else
+            // Search for the supported pki type strings
+            std::string pr = r.second;
```

Comment:
Please, factor this out to a function at least :smile:

Reply:
Put it into a function and cleaned up this for loop and if block a bit.

Reply:
thanks !